persistence-api/src/main/java/io/stargate/db/schema/SchemaHash.java

```diff
@@ -0,0 +1,69 @@
+package io.stargate.db.schema;
+
+import java.util.Collection;
+import java.util.Objects;
+
+/** An interface for calculating deterministic hash codes for schema objects. */
+public interface SchemaHash {
+
+  /**
+   * Calculates a deterministic hash code for the schema object that does not change between
+   * invocations of the JVM. This was mostly because it was determined that hash codes of
+   * enums could change between different executions of the JVM and schema hash codes need to
+   * be deterministic across JVM instances because these codes are distributed to Stargate
+   * bridge clients.
+   *
+   * <p>When calculating the schema hash code it advised that one of the `SchemaHash#hashCode()`
+   * methods in this interface be used instead of using `obj.hashCode()`, {@link Object#hashCode()},
+   * or {@link System#identityHashCode(Object)}, etc. to avoid an error that produces a
+   * non-deterministic hash code.
+   *
+   * @return a deterministic hash code for the schema object.
+   */
+  int schemaHashCode();
+
+  static int hashCode(SchemaHash object) {
+    if (object == null) {
+      return 0;
+    }
+    return object.schemaHashCode();
+  }
+
+  static int hashCode(Object object) {
+    // It's still possible that something like `Optional<Enum>` would make it through. So that's
+    // something to look out for.
+    if (object instanceof Enum || object instanceof SchemaHash) {
+      throw new AssertionError(
+          "Using `SchemaHash#hashCode(Object object)` on this object type (`Enum`, `SchemaHash`) may produce an non-deterministic hash code");
+    }
+    return Objects.hashCode(object);
+  }
+
+  static int hashCode(int i) {
+    return Integer.hashCode(i);
+  }
+
+  static int hashCode(boolean b) {
+    return Boolean.hashCode(b);
+  }
+
+  static int enumHashCode(Enum<?> object) {
+    if (object == null) {
+      return 0;
+    }
+    // FIXME: Is this good enough of a hash?
+    return object.ordinal() + 1; // Add one so that it's not ambiguous with null
```

Comment:
So ordinals do change, but names don't.. Use `object.name()`? Would protect against reordering, but this is good imo.. We don't have to have consistent hashes between app versions,,

Reply:
I agree that `object.name()` is better because it would keep the hashes consistent if a new entry is added or removed.